rusk-wallet/src/lib/clients.rs

```diff
@@ -202,21 +204,25 @@ impl ProverClient for Prover {
 }
 
 /// Implementation of the StateClient trait from wallet-core
-#[derive(Debug)]
 pub struct State {
     client: Mutex<GrpcStateClient<Channel>>,
     gql_url: String,
+    cache: Connection,
 }
 
 impl State {
-    pub fn new(client: GrpcStateClient<Channel>, gql_url: String) -> Self {
+    pub fn new(
+        client: GrpcStateClient<Channel>,
+        gql_url: String,
+        cache: Connection,
```

Comment:
I personally don't like this approach. It feels odd to have an SQL connection here in my humble opinion. I'd rather have the `cache` object, properly instantiated and with an owned `Connection` within it than passing a connection around. I think the backend of the cache should be totally opaque to the rest of the crate.

Reply:
you're right